### DISCUSSION TITLE
resx files / dependency list

### DISCUSSION THREAD

**Discussion by Kenney-code** [vscode-abl/vscode-abl#128] [Q&A] [answered]
Hi,
I've been converting some of my projects from pds to vscode.
However, it seems that I am lacking a visual designer that auto creates a .resx file for .net based classes.
How do you start the process of creating the necessary resx files from a progress class containing .net components?

Furthermore, it would be nice to grant a dependency list to the openedge project for custom assemblies or am I just missing how this should be done? (currently I am using a startup procedure to load my assemblies)

Kind regards

**Comment by PeterJudgeZA** [comment, marked as answer]
> Furthermore, it would be nice to grant a dependency list to the openedge project for custom assemblies or am I just missing how this should be done? (currently I am using a startup procedure to load my assemblies)

You can use the `-assemblies` startup parameter to specify the location of any assemblies that are not in the GAC. You can specify any startup parameters in the `extraParameters` property in the project's openedge-project.json file.

**Comment by gquerret** [comment]
> However, it seems that I am lacking a visual designer that auto creates a .resx file for .net based classes.
> How do you start the process of creating the necessary resx files from a progress class containing .net components?

There's currently no way to generate the resx files, and there are no plans to add a visual designer to VSCode.

**Reply by Kenney-code** [reply]
Are there any plans as to adding something that would be able to maintain resx files ?

**Reply by gquerret** [reply]
I have to admit that my knowledge of resx files is close to zero. What does it take to maintain those files ? @PeterJudgeZA Any idea ?

**Reply by PeterJudgeZA** [reply]
There are a variety of RESX editor extensions in VSCode. These allow editing of the .resx file. But I suspect that the updating of those files from the OpenEdge GUI for .NET classes (forms, controls etc) can only be done from the PDSOE Visual Designer.
